xapian-letor/ranker/svmranker.cc

```diff
@@ -45,6 +45,20 @@ Proceedings of the eighth ACM SIGKDD international conference on Knowledge disco
 using namespace std;
 using namespace Xapian;
 
+static void
+clear_svm_problem(svm_problem problem, int n)
+{
+    delete [] problem.y;
+    problem.y = NULL;
+    for (int i = 0; i < n;i++)
+    {
```

Comment:
Yes .... this change was quite small .... so I decided to review it manually .... it shows the fundamental wrong in my thinking while writing code .... I should have actually by now adapted to using xapian conventions like having a { on the same line as for, if and while and using prefix increment instead of postfix .... but I always do rush in pushing commits and keep making these mistakes for which I again apologize